Strategy Design Pattern for color themes and fixing tests

RawdaWaez opened the pull request:
The app supports 4 colour themes: light, Dark, Light for colour blind, and dark for colour blind and this has been implemented using a strategy design pattern.

<img width="401" alt="image" src="https://github.com/user-attachments/assets/845130f4-54b1-4050-ae63-14e1cbec21cd" />

<img width="384" alt="image" src="https://github.com/user-attachments/assets/986e1a13-8619-4258-bb40-8cef895f55a3" />

<img width="389" alt="image" src="https://github.com/user-attachments/assets/5c996c0a-6a5f-44b8-b1e0-533bcc297d6a" />

<img width="401" alt="image" src="https://github.com/user-attachments/assets/5726711e-8146-43d2-9027-fb6f20df0cc0" />

<img width="408" alt="image" src="https://github.com/user-attachments/assets/92bb2d3c-8f20-4bff-a6be-eaed7c44f32a" />

RawdaWaez committed:
Refactoring remove dead code.

RawdaWaez committed:
Implement the shuttle the dark/colorblind theme.

RawdaWaez committed:
Merge pull request #334 from Aymane-Arfaoui/caline

Caline to rawda

RawdaWaez committed:
Merge remote-tracking branch 'origin/AymaneDirectionsChat' into testrawdaa

# Conflicts:
#	app/services/chatService.js

RawdaWaez committed:
Merge remote-tracking branch 'origin/testrawdaa' into testrawdaa

RawdaWaez committed:
fix some minor errors.

RawdaWaez committed:
Revert "fix some minor errors."

This reverts commit f8a62e462ba5d26355a14ccdf0d95aecbd9a0af0.

RawdaWaez committed:
fix minor

RawdaWaez committed:
Merge remote-tracking branch 'origin/main' into testrawdaa

# Conflicts:
#	app/placeholder.jsx

RawdaWaez committed:
Implement the searchbars with the theme.

AbdelmalekMouhamou committed:
fixed failing tests

AbdelmalekMouhamou committed:
small fix to distance calcluation and to sonarQube.yml

AbdelmalekMouhamou committed:
added accessibility routing option

AbdelmalekMouhamou committed:
Refactored MapScreen

-many react hooks were run conditionally in MapScreen (depending on the validity of the buildingkeys provided in the route params)
-extracted the code that is ran conditionally into an inner-component and wrapped it with a components that makes the conditional check.
-removed argument that was provided to a function (closeIndoorSearchBars) that expects no arguments

RawdaWaez committed:
Implement Strategy Design pattern for the color themes.

atahle committed:
Maestro Testing Final Phase

atahle committed:
Merge branch 'testrawdaa' of https://github.com/Aymane-Arfaoui/FindMyClass into testrawdaa

atahle committed:
Fixing Mic problem related to UT-342

RawdaWaez committed:
Implement dark theme in floor selector and indoor searchbar.

AbdelmalekMouhamou committed:
added tests

atahle committed:
More maestro test for AI help for tasks

RawdaWaez committed:
Implement dark theme for the indoor components.

RawdaWaez committed:
Implement dark theme calendar.

AbdelmalekMouhamou committed:
fix home test

AbdelmalekMouhamou committed:
reliability improvements

AbdelmalekMouhamou committed:
removed console log to pass lint check

AbdelmalekMouhamou committed:
fixed useMemo called outside of component

RawdaWaez committed:
Implement dark theme for map screen.

AbdelmalekMouhamou committed:
small fix mapScreen

AbdelmalekMouhamou committed:
refactored homemap.jsx

refactored ferching functions in homemap to a fetching service and added its test

RawdaWaez committed:
Refactoring Smart Planner extract method and class.

RawdaWaez committed:
Merge remote-tracking branch 'origin/testrawdaa' into testrawdaa

AbdelmalekMouhamou committed:
added more test and quick fix to mapScreen

RawdaWaez committed:
Merge remote-tracking branch 'origin/main' into testrawdaa

# Conflicts:
#	.github/workflows/SonarQube.yml
#	app/services/chatService.js
#	app/shuttleSchedule.jsx
#	components/AppNavigationPannel.jsx
#	components/Calendar.jsx

Domat99 commented on app/chat.jsx:
```diff
@@ -1,17 +1,17 @@
+//       <ChatInterface navigation={{ goBack: () => router.back() }} />
+//     </>
+//   );
+// }
```
Why is the chat commented out here?

RawdaWaez replied:
I will fix it

AbdelmalekMouhamou committed:
small fix

moved fetchSmartPlanner, add type check to calendarService and prop Check to user.jsx

AbdelmalekMouhamou committed:
small fix

AbdelmalekMouhamou committed:
moved chat.py from /api to /api/app

AbdelmalekMouhamou committed:
fixed failing test

AbdelmalekMouhamou committed:
Update BottomPanel.test.jsx

RawdaWaez committed:
finalizing UI + fix errors related to pr

RawdaWaez committed:
Merge remote-tracking branch 'origin/testrawdaa' into testrawdaa

RawdaWaez committed:
minor coloring issues.

Domat99 commented on app/__tests__/_layout.test.jsx:
```diff
@@ -1,30 +1,36 @@
+    //
+    //    await waitFor(()=>{expect(replace).toHaveBeenCalled();});
+    //
+    // });
```
Please remove dead code here if not needed

RawdaWaez replied:
Good point! I have removed the dead code.

Domat99 commented on app/__tests__/homemap.test.jsx:
```diff
@@ -72,44 +74,64 @@ jest.mock('@/services/routeService', () => ({
-        //         //const noDetailsMessage = screen.queryByText(/Error fetching building details/i);
-        //         expect(mockConsole).toBeCalled();
-        //     });
-        // });
```
Good refactoring here by removing dead code!

Domat99 commented on constants/themeStrategies.js:
```diff
@@ -0,0 +1,22 @@
+
+export const ColorBlindDarkTheme = {
+    getColors: () => colorBlindDarkColors,
+};
```
Good job on implementing the Strategy Pattern!

Domat99 commented on services/fetchSmartPlanner.js:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    const storedEvents = await AsyncStorage.getItem("@calendar");
+    const storedTasks = await AsyncStorage.getItem("tasks");
```
It would be safer to surround the AsyncStorage with try/catch since it can fail.

RawdaWaez replied:
Done 

<img width="1041" alt="image" src="https://github.com/user-attachments/assets/99316112-a0cb-4d9f-bb22-0497bcd4e686" />

Domat99 commented on services/fetchingService.js:
```diff
@@ -0,0 +1,329 @@
+        }
+    } catch (error) {
+        setRoutes([]);
+        setFastestRoute(null);
```
It would be helpful to print the error message here.

RawdaWaez replied:
Done! 

<img width="941" alt="image" src="https://github.com/user-attachments/assets/7ccdd469-17dd-4a41-ad7c-f56240c0f9d6" />

Aymane-Arfaoui commented on context/ThemeProvider.js:
```diff
@@ -1,74 +1,69 @@
+                    AsyncStorage.getItem('@colorBlindMode'),
+                    AsyncStorage.getItem('@user'),
+                ]);
+
```
In the loadThemePreferences function. You might want to consider handling the scenario where storedTheme or storedCBMode might not be available.

If the AsyncStorage.getItem call fails, providing meaningful fallback values instead of just logging the error woud be better

RawdaWaez replied:
Good catch! I added default fallbacks and ensured both theme and color blind mode are always initialized properly, even if the values are missing in AsyncStorage.

<img width="844" alt="image" src="https://github.com/user-attachments/assets/20d179d4-5d26-4fd7-8399-9f4f21c04d7a" />

AbdelmalekMouhamou committed:
refactored mapscreen functions into a helperFile and tested it

RawdaWaez committed:
minor fixes related to pr review.

RawdaWaez committed:
minor fixes colors
